### PR TITLE
Obstacles Avoidance

### PR DESCRIPTION
**Pull request by Sigbear**
Added Obstacle recognition for agents through tag filtering on "Obstacle".
Added "Terrain" tag to every mesh handler gameObject being generated so we can fiter this out in the main flock context filter so units don't try to avoid the terrain itself. (This was done so we can pass down more than just players to relevant behaviors).
Because of this flock context is no longer filtered by "Players" tag. (This is done in later filter-behaviors)
Removed *Boundary* Tag for now.

### CODE REVIEW

File: KA-Flocking/Assets/Scripts/Filter Scripts/ObstacleFilter.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public override List<Transform> Filter(FlockAgent agent, List<Transform> original)
+    {
+        List<Transform> filtered = new List<Transform>();
+        foreach (var item in original)
+        {
+            if (item.CompareTag("Obstacle"))
+            {
+                Debug.Log("Collider");
+                Collider collider = item.GetComponent<Collider>();
+                Debug.Log(collider);
```
Comment (Hoidi) [inline KA-Flocking/Assets/Scripts/Filter Scripts/ObstacleFilter.cs]: Remove please. I don't want console spam

File: KA-Flocking/Assets/Scripts/Filter Scripts/ObstacleFilter.cs [hunk 2]
```diff
@@ -12,9 +12,9 @@ public override List<Transform> Filter(FlockAgent agent, List<Transform> origina
         {
             if (item.CompareTag("Obstacle"))
             {
-                Debug.Log("Collider");
+                // Debug.Log("Collider");
                 Collider collider = item.GetComponent<Collider>();
-                Debug.Log(collider);
+                // Debug.Log(collider);
```
Comment (Hoidi) [inline KA-Flocking/Assets/Scripts/Filter Scripts/ObstacleFilter.cs]: My dude. No comments :grin: Just remove them :+1:

Comment (Sigbear) [reply KA-Flocking/Assets/Scripts/Filter Scripts/ObstacleFilter.cs]: Hoidi pls

File: KA-Flocking/Assets/Scenes/FlockScene.unity [hunk 3]
```diff
@@ -38,7 +38,7 @@ RenderSettings:
   m_ReflectionIntensity: 1
   m_CustomReflection: {fileID: 0}
   m_Sun: {fileID: 0}
-  m_IndirectSpecularColor: {r: 0.44657826, g: 0.49641263, b: 0.57481676, a: 1}
+  m_IndirectSpecularColor: {r: 0.44657874, g: 0.49641258, b: 0.574817, a: 1}
```
Comment (jerge) [inline KA-Flocking/Assets/Scenes/FlockScene.unity]: Please don't do this, it leads to merge conflicts

Comment (jerge) [inline KA-Flocking/Assets/Scenes/FlockScene.unity]: Please don't do this, it leads to merge conflicts

Comment (jerge) [inline KA-Flocking/Assets/Scenes/FlockScene.unity]: Please don't do this, it leads to merge conflicts

**Comment by maxarfvidssonnilsson** [comment]
Maybe add walls to the edge of the generated world so that it is clear that it works.

**Pull request closed without merging**